Authorization/Status.ts

```diff
@@ -0,0 +1,47 @@
+import { gracely } from "gracely"
+import { isly } from "isly"
+import { Transaction } from "../Transaction"
+
+export type Status = "approved" | Status.Failed
+
+export namespace Status {
+	export interface Failed {
+		code: string
+		reason: string | string[]
+		error?: gracely.Error
+	}
+	export namespace Failed {
+		export function from(error: gracely.Error): Failed {
+			let result: Status
+			if (error.error?.includes("Card with id"))
```

Comment:
Is there a better way to deduce the error than from the error message, as it may be changed?